src/use-pushowl.ts

```diff
@@ -1,137 +1,135 @@
-import { useEffect, useState } from 'react'
-import { useCartState } from 'frontend-checkout'
-import { useCustomerState } from 'frontend-customer'
-import { StorePlatformDomain, Pushowl } from './types'
+import { useEffect, useState } from "react";
+import { StorePlatformDomain, Pushowl } from "./types";
 
 const isRecord = (value: unknown): value is Record<PropertyKey, unknown> =>
-    typeof value === 'object' && value !== null && !Array.isArray(value)
+  typeof value === "object" && value !== null && !Array.isArray(value);
 
 const getVariantIdFromItem = (item: any): string => {
-    // eslint-disable-next-line no-prototype-builtins
-    if (isRecord(item) && isRecord(item.variant) && item.variant.hasOwnProperty('id')) {
-        return `${item.variant.id}`
+  // eslint-disable-next-line no-prototype-builtins
+  if (
+    isRecord(item) &&
+    isRecord(item.variant) &&
+    item.variant.hasOwnProperty("id")
+  ) {
+    return `${item.variant.id}`;
+  }
+
+  // eslint-disable-next-line no-prototype-builtins
+  if (isRecord(item) && item.hasOwnProperty("variant_id")) {
+    return `${item.variant_id}`;
+  }
+
+  return "";
+};
+
+export const processCart = (taskInput: {
+  cartId: string;
+  cartItems: Array<any>;
+}) => {
+  const { cartItems, cartId } = taskInput;
+
+  const items = cartItems.map((item) => {
+    const productWithCheckout = atob(item.id).split("/").pop();
+
+    let productId = null;
+
+    if (productWithCheckout !== undefined && productWithCheckout.length > 0) {
+      productId = parseInt(productWithCheckout.split("?checkout=")[0], 10);
```

Comment:
hope this is reliable enough and doesn't have edge cases where `split` can fail

Reply:
Yes, it doesn't fail.